scripts/tests/test_safe_template_linter.py

```diff
@@ -721,12 +726,195 @@ def test_check_underscore_file_disables_one_violation(self):
     )
     def test_check_underscore_no_escape_allowed(self, data):
         """
-        Test _check_underscore_file_is_safe with expressions that are allowed
+        Test check_underscore_file_is_safe with expressions that are allowed
         without escaping because the internal calls properly escape.
         """
         linter = UnderscoreTemplateLinter()
         results = FileResults('')
 
-        linter._check_underscore_file_is_safe(data['template'], results)
+        linter.check_underscore_file_is_safe(data['template'], results)
 
         self.assertEqual(len(results.violations), 0)
+
+
+@ddt
+class TestJavaScriptLinter(TestLinter):
```

Comment:
@andy-armstrong @efischer19 @nedbat: FYI: I'd start by reviewing the tests to understand what is being linted for.